### ISSUE TITLE
[Feature] [Placement Group] Add timeout mechanism when scheduling placement group

### ISSUE THREAD

**Issue by clay4megtr** [ray-project/ray#20477]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

Currently, The scheduling of placement groups is one by one and there is no timeout mechanism when scheduling is in process, so, when there is something wrong with the worker node or the prepare/commit msg lost, the scheduling will hang for a long time or forever.

### Use case

N/A

### Related issues

N/A

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by rkooo567** [comment]
You mean timeout in prepare / commit rpcs?

**Comment by clay4444** [comment]
> You mean timeout in prepare / commit rpcs?

Yes,  we encountered this issue in the production env several days ago, it costs 10+ minutes to commit a bundle...   so we can just manually remove the worker node at that time.

**Comment by clay4444** [comment]
actually, no, I believe this was caused by the problematic node because there were also failures when creating workers on that node, but I didn't find the root cause.....  I even suspected that was a one-way network unreachable issue afterward.

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
